.buildkite/scripts/common.sh

```diff
@@ -146,3 +146,24 @@ check_is_arm() {
     is_arm=""
   fi
 }
+
+git_diff() {
+  patterns=$1
+  patterns_formatted="$(echo "$patterns" | tr ':' '\n')"
+
+  if git diff --name-only HEAD@{1} HEAD | grep -qE -e "$patterns_formatted"; then
```

Comment:
I wonder if I am missing something, but I wonder how is this supposed to work when one file doesn't match from a list in `$patterns`.

Consider the following:

```
$ git diff --name-only HEAD@{1} HEAD 
.buildkite/aarch64_pipeline.yml
.buildkite/linux_jdk_matrix_pipeline.yml
.buildkite/scripts/dra/generatesteps.py
.buildkite/scripts/jdk-matrix-tests/generate-steps.py
.buildkite/supported_plugins_test_pipeline.yml
.buildkite/windows_jdk_matrix_pipeline.yml
```

if I set `$patterns` to include a few of the above files it seems to be working as expected:

```
$ patterns=".buildkite/aarch64_pipeline.yml:.buildkite/linux_jdk_matrix_pipeline.yml"
$ patterns_formatted="$(echo "$patterns" | tr ':' '\n')"
$ git diff --name-only HEAD@{1} HEAD | grep -qE -e "$patterns_formatted"
$ echo $?
0
```

now if I introduce a deliberate typo to one of the paths, it still reports true:

```
$ patterns=".buildkite/aarch64_pipeline.yml:.buildkite/linux_jdk_matrix_pipeline.ymlTYPOHERE"
$ patterns_formatted="$(echo "$patterns" | tr ':' '\n')"
$ echo "$patterns_formatted"
.buildkite/aarch64_pipeline.yml
.buildkite/linux_jdk_matrix_pipeline.ymlTYPOHERE
$ git diff --name-only HEAD@{1} HEAD | grep -qE -e "$patterns_formatted"
$ echo $?
0
```

Is this the expected behavior?

Reply:
Great point, thanks! Adjusted logic, but had to move filechange list to separate files, since [BK can't manage regexes in environment variables](https://github.com/buildkite/agent/issues/1381).  But it'd not that bad, since it's easier to read/maintain separate file.